### ISSUE TITLE
convert -extent CMYK tiff broken in 6.9.10.68-3

### ISSUE THREAD

**Issue by qing917** [ImageMagick/ImageMagick#2621]
The issue happens after I update from 6.7.8.9-18 to 6.9.10.68-3.

With 6.7.8.9-18, I can extent a 1x1 (1 pixel) CMYK tiff file into 3x1 (3 pixel) CMYK tiff file, and fill the new pixels with cmyk(0,0,0,0) using:
`$convert test1x1.tif -gravity west -extent 3x1+0+0 test3x1.tif`
Please find the input 1x1 CMYK tiff file: [test1x1.tif.gz](https://github.com/ImageMagick/ImageMagick/files/5279954/test1x1.tif.gz)

But now with 6.9.10.68-3, it produce a all black 3 pixel grayscale file.

input file:
$ identify test1x1.tif test1x1.tif TIFF 1x1 1x1+0+0 8-bit CMYK 348B 0.000u 0:00.001
output file using 6.9.10.68-3:
$ identify test3x1.tif test3x1.tif TIFF 3x1 3x1+0+0 8-bit Grayscale Gray 328B 0.000u 0:00.001
output file using 6.7.8.9-18:
sh-4.2$ identify test3x1.tif test3x1.tif TIFF 3x1 3x1+0+0 8-bit CMYK 370B 0.000u 0:00.001

When using the alternative command below to specify colorspace
`$convert test1x1.tif -gravity west -extent 3x1+0+0 -set colorspace CMYK test3x1.tif`

The output file is kept in CMYK, but it contains 3 random color pixels looks greyish in the output file, i.e. [cmyk(0,0,0,42), cmyk(0,0,0,18), cmyk(0,0,0,50)]
My expected output file is attached, with 3 pixels [cmyk(0,0,0,100), cmyk(0,0,0,0), cmyk(0,0,0,0)]
[k3x1.tif.gz](https://github.com/ImageMagick/ImageMagick/files/5280394/k3x1.tif.gz)

Please let me know if you need more information or a solution to the issue.  Thanks!

**Comment by fmw42** [comment]
try setting the background to black. The default is white.

convert test1x1.tif -gravity west -background black -extent 3x1+0+0 -set colorspace CMYK test3x1.tif

I get 3 black looking pixels using IM 6.9.11.30 Q16 Mac OSX Sierra

**Comment by qing917** [comment]
Thanks @fmw42 .  I've updated my initial issue description with an expected output, i.e. one K100 pixel, two CMYK blank pixels cmyk(0,0,0,0).

I've tried 
`convert test1x1.tif -gravity west -background black -extent 3x1+0+0 -set colorspace CMYK test3x1_4.tif`
and
`convert test1x1.tif -gravity west -fill "cmyk(0,0,0,0)" -extent 3x1+0+0 -set colorspace CMYK test3x1_5.tif`

Both give me 3 cmyk(0,0,0,0), that's not correct, as my original cmyk(0,0,0,100) pixel is erased.

Suspect later release has something broken in CMYK image handling.
